spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java

```diff
@@ -673,6 +673,41 @@ public void insertMention(Mentionable mention) {
     }
 
     /**
+     * Inserts a mention. This will not take any token into consideration. This method is useful
+     * when you want to insert a mention which doesn't have a token.
+     *
+     * @param mention {@link Mentionable} to insert a span for
+     */
+    public void insertMentionWithoutToken(Mentionable mention) {
+        // Setup variables and ensure they are valid
+        Editable text = getEditableText();
+        int index = getSelectionStart();
+        index = index > 0 ? index : 0;
+
+        // Insert the span into the editor
+        MentionSpan mentionSpan = new MentionSpan(getContext(), mention);
+        String name = mention.getPrimaryText();
+
+        detachTextWatcher();
+        text.replace(index, index, name);
+        text.setSpan(mentionSpan, index, index + name.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        Selection.setSelection(text, index + name.length());
+        ensureMentionSpanIntegrity(text);
+        attachTextWatcher();
+
+        // Hide the suggestions and clear adapter
+        if (mSuggestionsVisibilityManager != null) {
+            mSuggestionsVisibilityManager.displaySuggestions(false);
+        }
+
+        // Reset input method since text has been changed (updates mention draw states)
+        InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+        if (imm != null) {
+            imm.restartInput(this);
+        }
```

Comment:
This code has a lot in common with the previous function. Can you please extract the common functionality into a private function they can both use, avoiding duplicate code?